src/OrchardCore.Modules/OrchardCore.GitHub/Configuration/GitHubAuthenticationOptionsConfiguration.cs

```diff
@@ -41,17 +42,17 @@ public void Configure(AuthenticationOptions options)
         }
 
         // Register the OpenID Connect client handler in the authentication handlers collection.
-        options.AddScheme(GitHubDefaults.AuthenticationScheme, builder =>
+        options.AddScheme(GitHubAuthenticationDefaults.AuthenticationScheme, builder =>
```

Comment:
We should consider using the simpler generic API here 😃

Reply:
@kevinchalet can you please have a look at the last commit and see if this address both point you mentioned?

Reply:
Hum no, that's not what I had in mind: we actually deliberately avoid using the regular `.AddX()` extensions for all the ASP.NET Core authentication handlers - e.g the OpenIddict server and validation handlers or the MSFT OIDC client handler - to ensure invalid/missing configurations don't cause runtime exceptions when the options are read.

What I had in mind was just:

```csharp
options.AddScheme<GitHubAuthenticationHandler>(GitHubAuthenticationDefaults.AuthenticationScheme, GitHubAuthenticationDefaults.DisplayName);
```

Reply:
I see. I made the changes. Let me know if this is all we need.

Reply:
Looks good!

Since you're making changes to that package, maybe that would the right time to fix the  `GitHubAuthenticationSettings` inconsistency?

Unlike the OpenID module, that package defines a `GitHubAuthenticationSettings` class which is stored as regular OC settings (which is good) but is also used as an options instance that the `GitHubAuthenticationOptions` will import to populate its own options. It's completely useless, completely inconsistent with what we do elsewhere and brings no benefit at all: we should consider removing the `GitHubAuthenticationSettingsConfiguration` class and directly resolving the site settings from `GitHubAuthenticationOptionsConfiguration.Configure()`.

Reply:
Agreed. Done. 

I also went one step further and removed the `IGitHubAuthenticationService` as well.